corehq/apps/dump_reload/sql/dump.py

```diff
@@ -7,40 +7,79 @@
-    ('phonelog.UserEntry', UserIDFilter('user_id')),
-    ('ota.DemoUserRestore', UserIDFilter('demo_user_id', include_web_users=False)),
-    ('domain_migration_flags.DomainMigrationProgress', SimpleFilter('domain')),
-    ('products.SQLProduct', SimpleFilter('domain')),
-    ('sms.MessagingEvent', SimpleFilter('domain')),
-    ('sms.MessagingSubEvent', SimpleFilter('parent__domain')),
-    ('sms.PhoneNumber', SimpleFilter('domain')),
+APP_LABELS_WITH_FILTER_KWARGS_TO_DUMP = OrderedDict((iterator.model_label, iterator) for iterator in [
+    FilteredModelIteratorBuilder('locations.LocationType', SimpleFilter('domain')),
+    FilteredModelIteratorBuilder('locations.SQLLocation', SimpleFilter('domain')),
+    FilteredModelIteratorBuilder('form_processor.XFormInstanceSQL', SimpleFilter('domain')),
+    FilteredModelIteratorBuilder('form_processor.XFormAttachmentSQL', SimpleFilter('form__domain')),
+    FilteredModelIteratorBuilder('form_processor.XFormOperationSQL', SimpleFilter('form__domain')),
+    FilteredModelIteratorBuilder('form_processor.CommCareCaseSQL', SimpleFilter('domain')),
+    FilteredModelIteratorBuilder('form_processor.CommCareCaseIndexSQL', SimpleFilter('domain')),
+    FilteredModelIteratorBuilder('form_processor.CaseAttachmentSQL', SimpleFilter('case__domain')),
+    FilteredModelIteratorBuilder('form_processor.CaseTransaction', SimpleFilter('case__domain')),
+    FilteredModelIteratorBuilder('form_processor.LedgerValue', SimpleFilter('domain')),
+    FilteredModelIteratorBuilder('form_processor.LedgerTransaction', SimpleFilter('case__domain')),
+    FilteredModelIteratorBuilder('case_search.CaseSearchConfig', SimpleFilter('domain')),
+    FilteredModelIteratorBuilder('data_interfaces.AutomaticUpdateRule', SimpleFilter('domain')),
+    FilteredModelIteratorBuilder('data_interfaces.CaseRuleAction', SimpleFilter('rule__domain')),
+    FilteredModelIteratorBuilder('data_interfaces.CaseRuleCriteria', SimpleFilter('rule__domain')),
+    FilteredModelIteratorBuilder('data_interfaces.CaseRuleSubmission', SimpleFilter('domain')),
+    UniqueFilteredModelIteratorBuilder('data_interfaces.ClosedParentDefinition', SimpleFilter('caserulecriteria__rule__domain')),
+    UniqueFilteredModelIteratorBuilder('data_interfaces.CustomMatchDefinition', SimpleFilter('caserulecriteria__rule__domain')),
+    UniqueFilteredModelIteratorBuilder('data_interfaces.MatchPropertyDefinition', SimpleFilter('caserulecriteria__rule__domain')),
+    UniqueFilteredModelIteratorBuilder('data_interfaces.CustomActionDefinition', SimpleFilter('automaticupdaterule__domain')),
+    UniqueFilteredModelIteratorBuilder('data_interfaces.UpdateCaseDefinition', SimpleFilter('automaticupdaterule__domain')),
+    FilteredModelIteratorBuilder('data_interfaces.CreateScheduleInstanceActionDefinition', SimpleFilter('alert_schedule__domain')),
+    FilteredModelIteratorBuilder('data_interfaces.CreateScheduleInstanceActionDefinition', SimpleFilter('timed_schedule__domain')),
```

Comment:
we can consolidate the two statements that have `CreateScheduleInstanceActionDefinition` by just filtering on `caseruleaction__rule__domain`